File: 2015/08/solution.js
let stringCharacters = 0
let codeCharacters = 0
let encodedCharacters = 0

const processLine = (line) => {
  codeCharacters += line.length
  stringCharacters += eval(line).length
  encodedCharacters += `"${line.replace(/[\\"']/g, '\\$&').replace(/\u0000/g, '\\0')}"`.length
}

const report = () => {
  console.log(`Part 1: ${codeCharacters - stringCharacters}`)
  console.log(`Part 2: ${encodedCharacters - codeCharacters}`)
}

require('readline').
    createInterface({ input: require('fs').createReadStream('input.txt') }).
    on('line', processLine).
    on('close', report)

File: 2015/08/solution.rb
string_chars = 0
code_chars = 0
encoded_chars = 0

File.open('input.txt').each_line do |line|
  line = line.chomp
  code_chars += line.size
  string_chars += eval(line).size
  encoded_chars += line.dump.size
end

puts 'Part 1', code_chars - string_chars
puts 'Part 2', encoded_chars - code_chars